flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/expressions/ExpressionParser.scala

```diff
@@ -451,6 +452,28 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
     }
   }
 
+  lazy val tableFunctionCall: PackratParser[LogicalNode] =
+    functionIdent ~ "(" ~ repsep(expression, ",") ~ ")" ^^ {
+    case name ~ _ ~ args ~ _ => UnresolvedTableFunctionCall(name.toUpperCase, args)
```

Comment:
I think it would not hurt if we support `split(c) as (f1, f2)` for all expressions. This could also help in cases where we expand the projection like `pojo.flatten() as (f1, f2)`. I don't like the idea of a logical parser. It is already confusing enough for the user that we have Java API expression strings and SQL strings, we don't need another string syntax.
@fhueske what do you think?